File: pSet6/cash/cash.py
from cs50 import get_float


def main():
    change = get_change_owed("Changed owed: ")
    cents = dollars_to_cents(change)

    # Print the amount of cents as number of coins to return
    print(cents_to_coins(cents))


# Prompt user for an amount of change
def get_change_owed(prompt):
    while True:
        n = get_float(prompt)
        if n > 0:
            break
    return n


# Convert dollars to cents
def dollars_to_cents(amount):
    c = round(amount * 100)
    return c


# Always use the largest coin possible
# Keep track of coin used
def cents_to_coins(cents):
    # How many quarters in cents
    quarters = cents // 25
    # 1. Check how many quarters in cents
    # 2. How many dimes in the remaining amount of cents
    dimes = (cents % 25) // 10
    # 1. Check how many quarters & dimes in cents
    # 2.  How many nickels in the remaining amount of cents
    nickels = ((cents % 25) % 10) // 5
    # 1. Check how many quarters & dimes & nickels in cents
    # 2. The remainings are pennies  by default
    pennies = ((cents % 25) % 10) % 5

    # Return amount of coins in total
    return quarters + dimes + nickels + pennies


if __name__ == "__main__":
    main()

File: pSet1/cash/cash.c
// Ask the user how much change is owed
// Print out the minimum number of coins required to pay back the user
#include <cs50.h>
#include <stdio.h>
#include <math.h>

float change;
int cents;
int coins;

// Coins available based on american currency
int quarter = 25;
int dime = 10;
int nickel = 5;
int penny = 1;

int main(void) 
{
    // While change is negative
    // Prompt the user for how much change is owed
    while (change < 0)
    {
        change = get_float("Changed Owed: ");
    }
    
    // Round change amount to the nearest penny
    cents = round(change * 100);
    
    // Count amount of quarters, dimes, nickels and pennies held in cents
    // Add 1 to coins everytime 
    while (cents >= quarter)
    {
        cents -= quarter;
        coins++;
    }
    while (cents >= dime)
    {
        cents -= dime;
        coins++;
    }
    while (cents >= nickel)
    {
        cents -= nickel;
        coins++;
    }
    while (cents >= penny)
    {
        cents -= penny;
        coins++;
    }
    
    // Print minimum number of coins
    printf("%i \n", coins);
}